sphinx/application.py

```diff
@@ -123,10 +124,24 @@ def __init__(self, srcdir, confdir, outdir, doctreedir, buildername,
         self.enumerable_nodes = {}              # type: Dict[nodes.Node, Tuple[unicode, Callable]]  # NOQA
         self.html_themes = {}                   # type: Dict[unicode, unicode]
 
-        self.srcdir = srcdir
+        # validate provided directories
+        self.srcdir = abspath(srcdir)
+        self.outdir = abspath(outdir)
+        self.doctreedir = abspath(doctreedir or path.join(outdir, '.doctrees'))
         self.confdir = confdir
-        self.outdir = outdir
-        self.doctreedir = doctreedir
+        if self.confdir:
+            self.confdir = abspath(self.confdir)
+            if not path.isfile(path.join(self.confdir, 'conf.py')):
+                raise ConfigError("config directory doesn't contain a "
+                                  "conf.py file (%s)" % confdir)
+
+        if not path.isdir(self.srcdir):
+            raise ConfigError('Cannot find source directory (%s)' %
```

Comment:
+1 for checking directory existence and regulate abspath at `Sphinx` class.
But I feel `ConfigError` is not good for this case. It is used for configuration error.
So these validation should raise other error.

Reply:
This is different to the issue below. IMO, the source directory is part of the configuration. For example, calling `sphinx-build SRCDIR OUTDIR` *configures* Sphinx to read input from `SRCDIR` and write output to `OUTDIR`. However, as noted there, perhaps `ConfigError` is only for configuration file errors? If so, perhaps I should add a new `ConfigError` exception and rename the existing one `ConfigFileError`? The renamed exception can subclass the new one.

Reply:
>However, as noted below, perhaps you think that ConfigError is only for configuration file errors?

Yes. I understand the command line arguments are part of configuration in a broad sense. But , in narrow sense, it is not. In a broad sense, almost of user inputs are "configuration". But it is too ambiguous.

To be clearly, I propose you to add new exception like `ArgumentError`. I prefer to use different name for these errors.

Reply:
Done